compiler/rustc_errors/src/lib.rs

```diff
@@ -1292,6 +1294,17 @@ impl<'a> DiagCtxtHandle<'a> {
         self.create_err(err).emit()
     }
 
+    /// See [`DiagCtxtHandle::stash_diagnostic`] for details.
+    #[track_caller]
+    pub fn stash_err(
```

Comment:
I'd like to keep the stash API as minimal as possible. Please remove this and just do `stash`+`unwrap` at the call site. There are already a couple of call sites like that.